### ISSUE TITLE
Shoryuken ActiveJob "executions" is always 1

### ISSUE THREAD

**Issue by davidrichey** [ruby-shoryuken/shoryuken#656]
It seems the [executions](https://github.com/rails/rails/blob/2828d6b955ad732d05d0509420679331baf3950b/activejob/lib/active_job/core.rb#L29) attribute on the ActiveJob instance is always 1 no matter the number of retries/receives that have occurred. This field would be helpful for tasks you would never want to retry or know the number of retires that have occurred. SQS provides an "ApproximateReceiveCount" field that could be used to adjust the executions number, provided in Shoryuken::Message.data.attributes.

 
Example Instance:

```
#<AJob
 @arguments=["a6d2964149d2298f199372e70a8077e2"],
 @enqueued_at="2021-02-19T11:32:22Z",
 @exception_executions={},
 @executions=1,
 @job_id="c675437f-6470-4bc4-ad3a-2dedfed908ff",
 @locale="en",
 @priority=nil,
 @provider_job_id=nil,
 @queue_name="development-ordered.fifo",
 @serialized_arguments=nil,
 @sqs_send_message_parameters={},
 @timezone="UTC">
```

Example Shoryuken::Message:

```
#<Shoryuken::Message:0x00007f9579fc6be0
 @client=#<Aws::SQS::Client>,
 @data=
  #<struct Aws::SQS::Types::Message
   message_id="XXX",
   ...
   body=""
   attributes=
    {"SenderId"=>"AIDAVT6B5PAOCHTM4C7GW",
     "ApproximateFirstReceiveTimestamp"=>"1613734343008",
     "ApproximateReceiveCount"=>"8",
     "SentTimestamp"=>"1613734342934",
     "SequenceNumber"=>"18859860065500655872",
     "MessageDeduplicationId"=>"05c268bd4caae0c7dee813a5c359a2faf11a34c1090fc98d6c4c83a3501d8000",
     "MessageGroupId"=>"ShoryukenMessage"},
   md5_of_message_attributes="",
   message_attributes=
    {"shoryuken_class"=>
      #<struct Aws::SQS::Types::MessageAttributeValue
       string_value="ActiveJob::QueueAdapters::ShoryukenAdapter::JobWrapper",
       binary_value=nil,
       string_list_values=[],
       binary_list_values=[],
       data_type="String">}>,
 @queue_name="",
 @queue_url="">
```

Our use case: We have jobs we want to attempt once & only once. If that job fails or other interruptions happen (server fails, etc), we do not want to attempt this job again

**Comment by cjlarose** [comment]
Thanks for the bug report! I'll open a PR shortly.

> Our use case: We have jobs we want to attempt once & only once. If that job fails or other interruptions happen (server fails, etc), we do not want to attempt this job again

For this specific use case, consider setting a maximum receive count on a redrive policy for the queue. Of course, this only works if you want the receive-only-once behavior for all messages in the queue.

**Comment by cjlarose** [comment]
Fixed in #657 and released as part of shoryuken 5.2.0.
